core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -193,12 +192,9 @@ public SearchSourceBuilder(StreamInput in) throws IOException {
         storedFieldsContext = in.readOptionalWriteable(StoredFieldsContext::new);
         from = in.readVInt();
         highlightBuilder = in.readOptionalWriteable(HighlightBuilder::new);
-        int indexBoostSize = in.readVInt();
-        if (indexBoostSize > 0) {
-            indexBoost = new ObjectFloatHashMap<>(indexBoostSize);
-            for (int i = 0; i < indexBoostSize; i++) {
-                indexBoost.put(in.readString(), in.readFloat());
-            }
+        List<IndexBoost> boosts = in.readList(IndexBoost::new);
+        if (boosts.size() > 0) {
```

Comment:
True.
I was trying to preserve state where indexBoosts isn't set (null). But if we initialize indexBoosts with empty array, no need to worry.